src/ui/ui.js

```diff
@@ -205,4 +207,74 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         }
     });
 
+    /**
+     * Updates the order of each block model according to the order of the blockUi
+     * elements in the story content editing area
+     *
+     * @param {Component} that - an instance of sjrk.storyTelling.ui.storyUi
+     * @param {Object} completionEvent - the event to be fired upon successful completion
+     */
+    sjrk.storyTelling.ui.updateBlockOrder = function (that, completionEvent) {
+        var blockUis = that.reorderer.container.children();
+        var loopCounter = 0;
```

Comment:
This odd way of structuring the iteration will likely throw off a reader - why not write a conventional "for" loop iterating over blockUIs, or else write blockUIs.forEach(function (blockUi, blockIndex) { .... ?